apps/bfd-db-migrator/src/test/java/gov/cms/bfd/migrator/app/MigratorAppIT.java

```diff
@@ -0,0 +1,327 @@
+package gov.cms.bfd.migrator.app;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.fail;
+
+import gov.cms.bfd.migrator.util.DatabaseTestUtils;
+import gov.cms.bfd.migrator.util.DatabaseTestUtils.DataSourceComponents;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Stream;
+import javax.sql.DataSource;
+import org.awaitility.Awaitility;
+import org.awaitility.Duration;
+import org.awaitility.core.ConditionTimeoutException;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** Tests the migrator app under various conditions to ensure it works correctly. */
+public final class MigratorAppIT {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(MigratorApp.class);
+
+  // Enum for determining which flyway script directory to run a test against
+  private enum TEST_DIRECTORY {
```

Comment:
The naming case here is unusual. I would have expected TestDirectory. A suggestion - make this a parametrized enum or a map and include the location path here and simplify the usage in the method that builds up the environment for the process.

Reply:
Yup mixed up my enum class naming scheme with the values; I'll fix that and also make it parameterized with the path.